source/Octopus.Tentacle/Scripts/Kubernetes/RunningKubernetesJobScript.cs

```diff
@@ -0,0 +1,323 @@
+
+namespace Octopus.Tentacle.Scripts.Kubernetes
+{
+    public class RunningKubernetesJobScript : IRunningScript
+    {
+        static readonly AsyncLazy<string> BootstrapRunnerScript = new(async () =>
+        {
+            using var stream = typeof(RunningKubernetesJobScript).Assembly.GetManifestResourceStream("Octopus.Tentacle.Kubernetes.bootstrapRunner.sh");
+            using var reader = new StreamReader(stream!, Encoding.UTF8);
+            return await reader.ReadToEndAsync();
+        });
+
+        readonly IScriptWorkspace workspace;
+        readonly ScriptTicket scriptTicket;
+        readonly string taskId;
+        readonly ILog log;
+        readonly IScriptStateStore stateStore;
+        readonly IKubernetesJobService jobService;
+        readonly IKubernetesClusterService kubernetesClusterService;
+        readonly KubernetesJobScriptExecutionContext executionContext;
+        readonly CancellationToken scriptCancellationToken;
+        readonly string? instanceName;
+        readonly KubernetesJobOutputStreamWriter outputStreamWriter;
+
+        public int ExitCode { get; private set; }
+        public ProcessState State { get; private set; }
+        public IScriptLog ScriptLog { get; }
+
+        public RunningKubernetesJobScript(
+            IScriptWorkspace workspace,
+            IScriptLog scriptLog,
+            ScriptTicket scriptTicket,
+            string taskId,
+            CancellationToken scriptCancellationToken,
+            ILog log,
+            IScriptStateStore stateStore,
+            IKubernetesJobService jobService,
+            IKubernetesClusterService kubernetesClusterService,
+            IApplicationInstanceSelector appInstanceSelector,
+            KubernetesJobScriptExecutionContext executionContext)
+        {
+            this.workspace = workspace;
+            this.scriptTicket = scriptTicket;
+            this.taskId = taskId;
+            this.log = log;
+            this.stateStore = stateStore;
+            this.jobService = jobService;
+            this.kubernetesClusterService = kubernetesClusterService;
+            this.executionContext = executionContext;
+            this.scriptCancellationToken = scriptCancellationToken;
+            ScriptLog = scriptLog;
+            instanceName = appInstanceSelector.Current.InstanceName;
+
+            outputStreamWriter = new KubernetesJobOutputStreamWriter(workspace);
+        }
+
+        public async Task Execute(CancellationToken taskCancellationToken)
+        {
+            var exitCode = -1;
+
+            var linkedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(scriptCancellationToken, taskCancellationToken);
+            var cancellationToken = linkedCancellationTokenSource.Token;
+            try
+            {
+                using var writer = ScriptLog.CreateWriter();
+                try
+                {
+                    using (ScriptIsolationMutex.Acquire(workspace.IsolationLevel,
```

Comment:
This presumably works because when running one instance, the mutex is accessable by that one Tentacle process.
If/when we allow multiple instances to run within a cluster we may need to answer if this isolationlevel check is still relevant.

Reply:
Correct. When/if we get to HA, then we'll need a different approach to how to do the script isolation mutex or, as you say, if it's still relevant